okhttp-clients/src/main/java/com/palantir/remoting3/okhttp/AsyncQosIoExceptionHandler.java

```diff
@@ -56,11 +68,12 @@
 
             @Override
             public ListenableFuture<Response> visit(QosException.Unavailable exception) {
-                if (!backoffStrategy.nextBackoff().isPresent()) {
+                Optional<Duration> backoff = backoffStrategy.nextBackoff();
+                if (!backoff.isPresent()) {
                     return Futures.immediateFailedFuture(qosIoException);
                 } else {
-                    // TODO(rfink): Use duration and schedule retry for later.
-                    return executorService.submit(() -> call.clone().execute());
+                    return executorService.schedule(
+                            () -> call.clone().execute(), backoff.get().toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
Consider adding the QoS exception as a suppressed exception if the cloned call fails

Reply:
How would you imagine this should work? I could inject it into the QosIoExceptionAwareCall call when cloning, but is that really worth it?

Reply:
Yes, that'd work. It might be useful when things are going south to be able to see how the service was responding before the client gave up, and how the client reacted. Up to you though.

Reply:
You'd probably need to timestamp the exceptions too

Reply:
OK, I think it'd be more manageable (and possibly useful) to log suitable information and piece things together with our log analysis tooling afterwards. Will add log statements liberally.